src/MCPServer/lib/unit.py

```diff
@@ -60,12 +60,20 @@ def reloadFileList(self):
             else:
                 files = File.objects.filter(sip_id=self.UUID)
             for f in files:
-                if f.currentlocation in self.fileList:
-                    self.fileList[f.currentlocation].UUID = f.uuid
-                    self.fileList[f.currentlocation].fileGrpUse = f.filegrpuse
+                try:
+                    file_unit = self.fileList.get(
+                        f.currentlocation,
+                        self.fileList.get(f.currentlocation.encode('utf8')))
+                except UnicodeError:
```

Comment:
I don't think it's possible for `KeyError` to be triggered here. `dict.get(key)` returns `None` if `key` is not in `dict` and `dict.get(key, dflt)` returns `dflt`.